### PR TITLE
Adding workaround for JDK > 8 invokedynamic tainting

### PR DESCRIPTION
**Pull request by oxeye-gal**
Hi,

We [@oxeye](https://oxeye.io) created a possible workaround for this previously discussed issue:

https://github.com/find-sec-bugs/find-sec-bugs/issues/575

We are not 100% sure that this is the optimal solution for this issue, so we would love to get your feedback on this so we can address your observations.

We also noticed that you have set the target version for fixing this issue for the next version (1.13 scheduled for December), is it still the due date?

Thanks

### CODE REVIEW

**Comment by oxeye-daniel** [comment]
@h3xstream - is there any update on this?

File: findsecbugs-plugin/src/main/java/com/h3xstream/findsecbugs/taintanalysis/TaintConfig.java [hunk 1]
```diff
@@ -216,6 +216,12 @@ private boolean isFieldType(String typeSignature) {
         return typeSignature != null && typeSignature.length() > 2 && typeSignature.charAt(0) != 'L';
     }
 
+    public int getMethodIdParamCount(String methodId) {
+        String signature = methodId.substring(methodId.indexOf("("), methodId.length());
+        SignatureParser p = new SignatureParser(signature);
+        return p.getNumParameters();
+    }
```
Comment (h3xstream) [inline findsecbugs-plugin/src/main/java/com/h3xstream/findsecbugs/taintanalysis/TaintConfig.java:223]: SignatureParser might not handle long and double parameters (which are double at runtime) but for `makeConcatWithConstants` it should be fine.

**Review by h3xstream — APPROVED** [review]
I didn't investigate deeply this PR but it should fix the missing method for StringConcatFactory / makeConcatWithConstants from Java 9.

**Comment by h3xstream** [comment]
Might fix #701

**Comment by jbindel** [comment]
It seems to help greatly for JDK11 bytecodes  with the new option turned on. One thing that does not get fixed is taint propagation for `makeConcatWithConstants` when concatenating a `long` or `double` with strings, which I believe is what you noted.

`"something" + 1000L` is tainted while `"something" + Long.toString(1000L)` is clean.

**Comment by jbindel** [comment]
I have some changes building on this pull request to make `TaintConfig` work when concatenating `long` and `double` values, and I'll make a pull request for them soon.

**Pull request merged**